lib/redux/inventory/actions.dart

```diff
@@ -18,8 +18,10 @@ ThunkAction<AppState> getInventory = (Store<AppState> store) async {
     'https://online.ntnu.no/api/v1/inventory/',
     headers: {'Authorization': 'Bearer ${token.accessToken}'},
   );
-  List jsonArray = json.decode(response.body);
-  List<Product> inventory =
-      jsonArray.map((product) => Product.fromJson(product));
-  await store.dispatch(SetInventory(inventory));
+  final List jsonArray = json.decode(response.body);
+  final List<Product> inventory =
+      jsonArray.map((product) => Product.fromJson(product)).toList();
+  return inventory != store.state.inventory
```

Comment:
Lists can't be compared with an euqals comparator and needs some more checking. Here is an example in Dart. https://stackoverflow.com/questions/10404516/how-can-i-compare-lists-for-equality-in-dart